### ISSUE TITLE
[Expand Mobile putFile RPC] SDL expected behavior

### ISSUE THREAD

**Issue by AByzhynar** [smartdevicelink/sdl_requirements#64]
Hello @lchen139 and @tpulatha
Please help to clarify some questions regarding proposal:
[0037-Expand-Mobile-putfile-RPC.md](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0037-Expand-Mobile-putfile-RPC.md
)
The proposal says:

> This paired with the ability to upload very large files as chunks using the offset and length parameters of putFile results in the problem that an App might upload a very large file (say 1GB) and gets a SUCCESS for each small putFile. But when the complete file gets CRC checked on the head unit, this check fails because one putFile had a missing frame that went unnoticed.

1) So in case of app sends very large file as chunks  - every small PutFile will send CRC checksum for every chunk, right? (Not for entire file). And in case of CRC checksum mismatch SDL core will send the `putFile` response with these values: 

> resultCode: CORRUPTED_DATA
> info: "CRC Check on file failed. File upload has been canceled, please retry."
> success: false

for every small PutFile ?

2) What if one of the chunks in the middle has failed CRC checksum but mobile app continues sending the rest of chunks and  the next ones were successful  - there is no sense to upload such file to the system. 
What SDL behaviour do you expect?

**Comment by AByzhynar** [comment]
@KhrystynaDubovyk Please be aware

**Comment by AByzhynar** [comment]
@lchen139 So please confirm that I am right in the following: 
```
App -> SDL : PutFile request (syncFileName, fileType, offset = 0, length, crc) - CRC check passed
SDL -> App : PutFile response (resultCode=SUCCESS, spaceAvailable) 
App -> SDL : PutFile request (syncFileName, fileType, offset = 100, length, crc) - CRC check passed
SDL -> App : PutFile response (resultCode=SUCCESS, spaceAvailable) 
App -> SDL : PutFile request (syncFileName, fileType, offset = 200, length, crc) - CRC check FAILED
SDL -> App : PutFile response (resultCode=CORRUPTED_DATA, spaceAvailable) 
```
In case app continues sending chunks:

```
App -> SDL : PutFile request (syncFileName, fileType, offset = 300, length, crc) - CRC check PASSED
SDL -> App : PutFile response (resultCode=INVALID_DATA, spaceAvailable) 
```
for all next PutFile requests with not null offset and not writing file to file system. Is this behavior correct and expected?

**Comment by lchen139** [comment]
@AByzhynar Yes, that aligns with my expectation. App putfile with wrong offset, SDL shall discard that incoming putfile(with wrong offset) and provide a negative response to notify app.

**Comment by AByzhynar** [comment]
@lchen139 Thank you very much
